### PR TITLE
Use hub's v2 APIs for creating users and teams in Cypress

### PR DESCRIPTION
**Pull request by vidyanambiar**
Swaps out the v1 APIs for creating/deleting users and teams with the new v2 APIs.

This can be tested with `remote-access.cy.ts`. This PR also un-skips this spec file.

Note there are still 4 remaining usages of `/api/galaxy/_ui/v1/users/` in `cypress/support/e2e.ts`. We can swap those out after the following API bug is addressed: [AAP-29429](https://issues.redhat.com/browse/AAP-29429)

### CODE REVIEW

**Review by vishsanghishetty — DISMISSED** [review]

File: cypress/e2e/hub/remotes-access.cy.ts [hunk 1]
```diff
@@ -1,17 +1,37 @@
...
 import { HubRemote } from '../../../frontend/hub/administration/remotes/Remotes';
+import { randomString } from '../../../framework/utils/random-string';
+import { ContentTypeEnum } from '../../../frontend/hub/interfaces/expanded/ContentType';
+import { HubRbacRole } from '../../../frontend/hub/interfaces/expanded/HubRbacRole';
 
-describe.skip('Remotes User Access tab', () => {
+describe('Remotes User Access tab', () => {
   let remote: HubRemote;
+  let role: HubRbacRole;
+  const customRole = {
```
Comment (nixocio) [inline cypress/e2e/hub/remotes-access.cy.ts:11]: Nice

File: cypress/e2e/hub/remotes-access.cy.ts [hunk 2]
```diff
@@ -44,18 +64,18 @@ describe.skip('Remotes User Access tab', () => {
...
+        cy.selectTableRowByCheckbox('name', role.name, {
           disableFilter: true,
         });
         cy.clickButton(/^Next/);
         cy.contains('h1', 'Review').should('be.visible');
         cy.verifyReviewStepWizardDetails('users', [hubUser.username], '1');
         cy.verifyReviewStepWizardDetails(
           'hubRoles',
-          ['galaxy.collection_remote_owner', 'Manage collection remotes.'],
+          [role.name, 'Manage collection remotes.'],
```
Comment (nixocio) [inline cypress/e2e/hub/remotes-access.cy.ts:78]:
```suggestion
          [role.name, role.description],
```

File: cypress/e2e/hub/remotes-access.cy.ts [hunk 3]
```diff
@@ -90,18 +110,18 @@ describe.skip('Remotes User Access tab', () => {
...
+        cy.selectTableRowByCheckbox('name', role.name, {
           disableFilter: true,
         });
         cy.clickButton(/^Next/);
         cy.contains('h1', 'Review').should('be.visible');
         cy.verifyReviewStepWizardDetails('teams', [hubTeam.name], '1');
         cy.verifyReviewStepWizardDetails(
           'hubRoles',
-          ['galaxy.collection_remote_owner', 'Manage collection remotes.'],
+          [role.name, 'Manage collection remotes.'],
```
Comment (nixocio) [inline cypress/e2e/hub/remotes-access.cy.ts:124]:
```suggestion
          [role.name, role.description],
```

File: cypress/support/hub-access-commands.ts [hunk 4]
```diff
@@ -5,7 +5,7 @@ import { HubTeam } from '../../frontend/hub/interfaces/expanded/HubTeam';
 
 Cypress.Commands.add('createHubUser', (hubUser?: Partial<HubUser>) => {
   cy.requestPost<HubUser, Partial<HubUser> & { username?: string; password?: string }>(
-    hubAPI`/_ui/v1/users/`,
+    hubAPI`/_ui/v2/users/`,
```
Comment (nixocio) [inline cypress/support/hub-access-commands.ts:8]: 👍

File: frontend/hub/administration/remotes/RemotePage/RemoteAddTeam.tsx [hunk 5]
```diff
@@ -144,7 +144,7 @@ export function RemoteAddTeams() {
   return (
     <PageLayout>
       <PageHeader
-        title={t('Add teams')}
+        title={t('Add roles')}
```
Comment (nixocio) [inline frontend/hub/administration/remotes/RemotePage/RemoteAddTeam.tsx:147]: Good catch.

**Review by nixocio — APPROVED** [review]
@vidyanambiar I left minor suggestions.

File: cypress/e2e/hub/remotes-access.cy.ts (hunk 2 above)
Comment (vidyanambiar) [reply cypress/e2e/hub/remotes-access.cy.ts]: good catch! updated 👍

File: cypress/e2e/hub/remotes-access.cy.ts (hunk 3 above)
Comment (vidyanambiar) [reply cypress/e2e/hub/remotes-access.cy.ts]: good catch! updated 👍

**Review by nixocio — APPROVED** [review]

**Pull request merged**